pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -306,6 +306,8 @@ func (r *RepoSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 			if updateErr != nil {
 				r.logger(ctx).Error(updateErr, "Sync status update failed")
 			}
+			// Use the deployment get error for metric tagging.
+			metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
```

Comment:
Should we just call this in a `defer`? It seems like we call this before every single return - or is the behavior more nuanced than that?

Reply:
I have a partial for that in https://github.com/GoogleContainerTools/kpt-config-sync/pull/592 which pulls out  upsertManagedObjects and deleteManagedObjects methods.

The gotcha with a defer is changing err to be at the reconcile scope and making sure all the child scopes populate the error in the parent scope, which can be hard to enforce/discover.